lib/core/presentation/widgets/poap/poap_claim_transfer_controller_widget/widgets/claim_modification_popup.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:app/core/domain/poap/entities/poap_entities.dart';
+import 'package:app/core/domain/poap/poap_enums.dart';
+import 'package:app/core/domain/token/entities/token_entities.dart';
+import 'package:app/core/presentation/widgets/common/dialog/lemon_alert_dialog.dart';
+import 'package:app/core/presentation/widgets/poap/poap_claimed_poup/poap_claimed_popup.dart';
+import 'package:app/core/presentation/widgets/poap/popap_busy_popup/poap_busy_popup.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:flutter/material.dart';
+
+class ClaimModificationPopup extends StatelessWidget {
+  final Claim? claimModification;
+  final TokenDetail? token;
+  final Function() onClose;
+  final Function(TokenDetail? token) onPressedTransfer;
+
+  const ClaimModificationPopup({
+    super.key,
+    this.claimModification,
+    this.token,
+    required this.onClose,
+    required this.onPressedTransfer,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    if (claimModification?.state == ClaimState.FAILED) {
+      final errorDescriptionName = claimModification?.errorDescription != null
+          ? claimModification?.errorDescription!['name']
+          : '';
+      String errorDescriptionMessage = '';
+      if (errorDescriptionName == ClaimErrorDescriptionName.AllClaimed.name) {
+        errorDescriptionMessage = t.nft.claimFailedDescription.allClaimed;
+      }
+      if (errorDescriptionName ==
+          ClaimErrorDescriptionName.AlreadyClaimed.name) {
+        errorDescriptionMessage = t.nft.claimFailedDescription.alreadyClaimed;
+      }
+
+      if (errorDescriptionName == ClaimErrorDescriptionName.Forbidden.name) {
+        errorDescriptionMessage = t.nft.claimFailedDescription.forbidden;
+      } else {
+        errorDescriptionMessage = errorDescriptionName ?? '';
+      }
```

Comment:
Too much if - else statements. Try to convert it into switch for more readable

Reply:
Also, avoid putting too many logic lines inside `build` method, you can extract it into a separate method

Reply:
For this case, I cannot use switch, it keeps saying this error, do you have any idea? 
<img width="807" alt="Screenshot 2023-10-12 at 09 16 39" src="https://github.com/lemonadesocial/lemonade-flutter/assets/28992487/020097d9-b2c2-493b-9224-dd5886e9243a">

Reply:
Then I think you just need to put the logic in a separate method. Switch case required constant value